### DISCUSSION TITLE
Next.JS-14 localStorage undefined issue

### DISCUSSION THREAD

**Discussion by qety4** [vercel/next.js#58246] [App Router]
### Link to the code that reproduces this issue

https://github.com/qety4/next14-localStorage-issue

### To Reproduce

Start a next app try to use localStorage in a separate hook/useLocalStorage hook in the src directory.

### Current vs. Expected behavior

I expect localStorage to be defined, like in the app I'm migrating from next 13.5.6

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
node 20.9.0, Mac iOS 12+
```

### Which area(s) are affected? (Select all that apply)

App Router

### Additional context

trying to use it for a react context, it works on next 13.5.6, however with the new next installation the app fails

**Comment by wlechowicz** [comment]
It's rendering your component on the server, where localStorage doesn't exist.
You can solve it like this, for example: `const savedJSON = global?.localStorage?.getItem(key);`
